patina_dxe_core/src/gcd.rs

```diff
@@ -329,6 +413,79 @@ pub(crate) fn activate_compatibility_mode() {
     crate::memory_attributes_protocol::uninstall_memory_attributes_protocol();
 }
 
+// ...existing code...
+// =============================================================================
+// ResourceDescriptor HOB Parsing Functions
+// =============================================================================
+// These functions provide clean separation between V1 and V2 ResourceDescriptor
+// HOB processing using conditional compilation. Only one version is compiled
+// based on the feature flags, ensuring zero runtime overhead and clean code paths.
+//
+// Architecture Benefits:
+// - Zero runtime branching: Feature selection happens at compile time
+// - Clean code paths: V1 and V2 logic completely separated
+// - Minimal duplication: Only parsing logic is separate, GCD logic is shared
+// - Future-proof: Easy to remove V1 support when no longer needed
+// =============================================================================
+
+/// Parse ResourceDescriptor HOB for V2 platforms (default, always compiled)
+///
+/// This function only processes V2 HOBs:
+/// - V2 HOBs: Preferred, provides cache attributes for optimal performance
+/// - V1 HOBs: Ignored completely (logs warning)
+///
+/// Returns: Some((ResourceDescriptor, cache_attributes)) or None if not a V2 resource descriptor
+#[cfg(not(feature = "v1_resource_descriptor_support"))]
+fn parse_resource_descriptor_hob(hob: &Hob) -> Option<(hob::ResourceDescriptor, u64)> {
+    match hob {
+        Hob::ResourceDescriptorV2(v2_res_desc) => {
+            // V2 platforms: Only process V2 HOBs
+            log::info!(
+                "Processing V2 Resource Descriptor HOB at {:#x} length {:#x} with cache attributes {:#x}",
+                v2_res_desc.v1.physical_start,
+                v2_res_desc.v1.resource_length,
+                v2_res_desc.attributes
+            );
+            Some((v2_res_desc.v1, v2_res_desc.attributes))
+        }
+        Hob::ResourceDescriptor(_) => {
+            // V2 platforms: Ignore V1 HOBs completely
+            log::warn!("Ignoring V1 Resource Descriptor HOB on V2-only platform");
```

Comment:
With my suggested change above mentioning we are skipping V1 HOBs, I don't think we have to print that for each V1 HOB.